yamlparse-applicative/src/YamlParse/Applicative/Pretty.hs

```diff
@@ -57,29 +73,32 @@ emptyComments = Comments []
 comment :: Text -> Comments
 comment t = Comments $ map pretty $ T.lines t
 
+
+data Color = Yellow | Grey
+
 -- | Prettyprint a 'Schema'
-schemaDoc :: Schema -> Doc ()
+schemaDoc :: Schema -> Doc Color
 schemaDoc = go emptyComments
   where
-    go :: Comments -> Schema -> Doc ()
+    go :: Comments -> Schema -> Doc Color
     go cs =
       let g = go cs
           ge = go emptyComments
-          mkComment :: Doc () -> Doc ()
-          mkComment = ("# " <>)
-          mkCommentsMDoc :: Comments -> Maybe (Doc ())
+          mkComment :: Doc Color -> Doc Color
+          mkComment = (annotate Grey) . ("# " <>)
+          mkCommentsMDoc :: Comments -> Maybe (Doc Color)
           mkCommentsMDoc = \case
             Comments [] -> Nothing
             Comments l -> Just $ align $ vsep $ map mkComment l
```

Comment:
This is where the actual rendering of the comments happens.
I think we should keep the `Doc` inside the `Comments` constructor un-annotated, and then only annotate it here when we get them out of this `Comments` constructor.
That way definitely all comments have been annotated correctl.

Reply:
Yeah I agree. I made some changes that should implement your suggestion